src/views/virtualmachines/actions/components/DeleteVMModal/components/GracePeriodInput.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  Checkbox,
+  Flex,
+  FlexItem,
+  InputGroup,
+  InputGroupText,
+  Popover,
+  StackItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import './grace-period-input.scss';
+
+const DEFAULT_GRACE_PERIOD = 180;
+
+interface GracePeriodInputProps {
+  gracePeriodSeconds: number | null;
+  setGracePeriodSeconds: (number) => void;
+}
+
+export const GracePeriodInput: FC<GracePeriodInputProps> = ({
+  gracePeriodSeconds,
+  setGracePeriodSeconds,
+}) => {
+  const { t } = useTranslation();
+
+  const onGracePeriodCheckChange = (checked: boolean) => {
+    if (checked) setGracePeriodSeconds(DEFAULT_GRACE_PERIOD);
+    else setGracePeriodSeconds(null);
+  };
+
+  const onGracePeriodSecondsChange = (value: string) => {
```

Comment:
this function might be having an issue when you try to delete the input, but the 0 keeps poping back.
also `parseInt` has a default of base 10, so you can omit the 10 at line 40.

Reply:
Yes. Otherwise, I would leave the input blank and handle the case of submission with the checkbox checked and input blank.